soh/soh/Enhancements/debugger/actorViewer.cpp

```diff
@@ -0,0 +1,1261 @@
+        case ACTOR_BG_BOWL_WALL:
+            return "Bombchu Bowling Alley Wall";
+        case ACTOR_EN_WALL_TUBO:
+            return "Bombchu Bowling Alley Bullseyes";
+        case ACTOR_EN_PO_DESERT:
+            return "Poe Guide (Desert Wasteland)";
+        case ACTOR_EN_CROW:
+            return "Guay";
+        case ACTOR_DOOR_KILLER:
+            return "Fake Door";
+        case ACTOR_BG_SPOT11_OASIS:
+            return "Oasis (Desert Colossus)";
+        case ACTOR_BG_SPOT18_FUTA:
+            return "Goron Jar Lid";
+        case ACTOR_BG_SPOT18_SHUTTER:
+            return "Sliding Doors (Goron City)";
+        case ACTOR_EN_MA3:
+            return "Malon (Adult, Lon Lon Ranch)";
+        case ACTOR_EN_COW:
+            return "Cow";
+        case ACTOR_BG_ICE_TURARA:
+            return "Icicles";
+        case ACTOR_BG_ICE_SHUTTER:
+            return "Vertical Ice Bars (Ice Cavern)";
+        case ACTOR_EN_KAKASI2:
+            return "Pierre the Scarecrow Spawn";
+        case ACTOR_EN_KAKASI3:
+            return "Bonooru the Scarecrow";
+        case ACTOR_OCEFF_WIPE4:
+            return "Scarecrow's Song Ocarina Effect";
+        case ACTOR_EN_EG:
+            return "Void-out Trigger (Tower Collapse)";
+        case ACTOR_BG_MENKURI_NISEKABE:
+            return "False Stone Walls (Gerudo Training Grounds)";
+        case ACTOR_EN_ZO:
+            return "Zora";
+        case ACTOR_OBJ_MAKEKINSUTA:
+            return "Skulltula Sprouting from Bean Spot";
+        case ACTOR_EN_GE3:
+            return "Gerudo Fortress Leader";
+        case ACTOR_OBJ_TIMEBLOCK:
+            return "Time Block";
+        case ACTOR_OBJ_HAMISHI:
+            return "Bronze Boulder";
+        case ACTOR_EN_ZL4:
+            return "Zelda (Child)";
+        case ACTOR_EN_MM2:
+            return "Running Man (Adult Era)";
+        case ACTOR_BG_JYA_BLOCK:
+            return "Silver Block (Child Era)";
+        case ACTOR_OBJ_WARP2BLOCK:
+            return "Navi Infospot (Green, Time Block)";
+        default:
+            return "";
+    }
+
+    return "";
+}
+
+template <typename T> void DrawGroupWithBorder(T&& drawFunc) {
+    // First group encapsulates the inner portion and border
+    ImGui::BeginGroup();
+
+    ImVec2 padding = ImGui::GetStyle().FramePadding;
+    ImVec2 p0 = ImGui::GetCursorScreenPos();
+    ImGui::SetCursorScreenPos(ImVec2(p0.x + padding.x, p0.y + padding.y));
+
+    // Second group encapsulates just the inner portion
+    ImGui::BeginGroup();
+
+    drawFunc();
+
+    ImGui::Dummy(padding);
+    ImGui::EndGroup();
+
+    ImVec2 p1 = ImGui::GetItemRectMax();
+    p1.x += padding.x;
+    ImVec4 borderCol = ImGui::GetStyle().Colors[ImGuiCol_Border];
+    ImGui::GetWindowDrawList()->AddRect(
+        p0, p1, IM_COL32(borderCol.x * 255, borderCol.y * 255, borderCol.z * 255, borderCol.w * 255));
+
+    ImGui::EndGroup();
+}
+
+std::vector<Actor*> PopulateActorDropdown(int i, std::vector<Actor*> data) {
```

Comment:
I'm pretty sure you can pass a reference to the vector and avoid returning a copy.